Functions/Quantification/xASL_quant_FSL.m

```diff
@@ -0,0 +1,569 @@
+    xASL_delete(pathFSLInput);
+	xASL_delete(pathFSLOptions);
+	xASL_delete(pathFSLOutput);
+	%xASL_adm_DeleteFileList(x.dir.SESSIONDIR, '(?i)^.*basil.*$', 1, [0 Inf]);
+    
+end
+
+function [BasilOptions] = xASL_sub_FSLOptions(pathFSLOptions, x, bUseFabber, PWI)
+%xASL_sub_FSLOptions generates the options and saves them in a file and returns some commandline options as well
+%
+% FORMAT: [BasilOptions] = xASL_sub_FSLOptions(pathFSLOptions, x, bUseFabber, PWI)
+% 
+% INPUT:
+%   pathFSLOptions  - filepath to the options file (REQUIRED)
+%   x               - struct containing pipeline environment parameters (REQUIRED)
+%   bUseFabber      - Use FABBER, alternative BASIL (REQUIRED)
+%
+% OUTPUT:
+% BasilOptions      - command-line options
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION: Options-file is saved and commandline options returned in a single string
+%
+% 1. Create the options file
+% 2. Basic model and tissue parameters
+% 6. Save and close the options file
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% EXAMPLE: [BasilOptions] = xASL_sub_FSLOptions(pathFSLOptions, x, bUseFabber, PWI)
+%
+% __________________________________
+% Copyright 2015-2023 ExploreASL 
+
+%% 0. Admin
+if ~bUseFabber
+	% Set BASIL dataPar options
+	if ~isfield(x.Q,'BASIL')
+		x.Q.BASIL = [];
+	end
+
+	if ~isfield(x.Q.BASIL,'bSpatial') || isempty(x.Q.BASIL.bSpatial)
+		fprintf('BASIL: Setting default option bSpatial = true\n');
+		x.Q.BASIL.bSpatial = true;
+	end
+
+	if ~isfield(x.Q.BASIL,'bInferT1') || isempty(x.Q.BASIL.bInferT1)
+		fprintf('BASIL: Setting default option bInferT1 = false\n');
+		x.Q.BASIL.bInferT1 = false;
+	end
+
+	if ~isfield(x.Q.BASIL,'bInferArt') || isempty(x.Q.BASIL.bInferArt)
+		fprintf('BASIL: Setting default option bInferArt = true\n');
+		x.Q.BASIL.bInferArt = true;
+	end
+
+	if ~isfield(x.Q.BASIL,'Exch') || isempty(x.Q.BASIL.Exch)
+		fprintf('BASIL: Setting default option Exch = simple\n');
+		x.Q.BASIL.Exch = 'simple';
+	end
+
+	if ~isfield(x.Q.BASIL,'Disp') || isempty(x.Q.BASIL.Disp)
+		fprintf('BASIL: Setting default option Disp = none\n');
+		x.Q.BASIL.Disp = 'none';
+	end
+
+end
+
+%% 1. Create the options file
+% BasilOptions is a character array containing CLI args for the Basil command
+BasilOptions = '';
+
+FIDoptionFile = fopen(pathFSLOptions, 'w+');
+if bUseFabber
+	fprintf(FIDoptionFile, '# Fabber options written by ExploreASL\n');
+else
+	fprintf(FIDoptionFile, '# Basil options written by ExploreASL\n');
+end
+
+%% 2. Basic model and tissue parameters
+% Basic model options
+if bUseFabber
+	fprintf(FIDoptionFile, '--method=vb\n');
+	fprintf(FIDoptionFile, '--model=asl_multite\n');
+	fprintf(FIDoptionFile, '--infertexch\n'); % Fit Tex
+	fprintf(FIDoptionFile, '--inferitt\n');   % Fit ATT
```

Comment:
is this correct?

Reply:
that's what Bea got from Amnah. And her output really provides Tex and ATT, so I guess that these are the correct names in FABBER. ButBea should of course confirm.

Reply:
@BeatrizPadrela ?